diddiparser2/parser.py

```diff
@@ -178,33 +177,63 @@ def parse_string_indexing(self, text):
             final_line = f"{final_line}{true_value}{index[1]}"
         return final_line
 
+    def resolve_value_or_variable(self, arg):
+        """
+        This resolves between direct values or variables
+        using the DSGP 2 specification.
+        """
+        if arg in EXECUTION_VARIABLES.keys():
+            return EXECUTION_VARIABLES[arg]
+        return self.identify_value(arg)
+
     def execute_func(self, line):
         "Run a call(argument) function."
         parsed_line = line.replace(");", "")
         call = parsed_line.split("(")[0]
         pos = len(f"{call}(")  # use this to avoid conflicts
-        arg = parsed_line[pos:]
-        del pos  # delete the aux
-        if arg.startswith("'") or arg.startswith('"'):
-            arg = arg[1:]
-        if arg.endswith("'") or arg.endswith('"'):
-            arg = arg[:-1]
-        arg = self.parse_string_indexing(arg)
+        args_aux = parsed_line[pos:]
+        args_aux = args_aux.split(",")
+        last_piece = ""
+        args = []
+        for piece in args_aux:
+            counter = None
+            if "'" in piece:
+                counter = 0
+                for char in piece:
+                    counter += 1 if char == "'" else 0
+            if '"' in piece:
+                counter = 0
+                for char in piece:
+                    counter += 1 if char == '"' else 0
+            if counter is not None:
+                if counter < 2 and (
+                    piece.strip().endswith("'") or piece.strip().endswith('"')
+                ):
+                    piece = last_piece + "," + piece
+                    args.pop()
+            last_piece = piece
+            args.append(piece)
+        fixed_args = []
+        del pos, args_aux, last_piece  # delete the aux
+        for arg in args:
+            if arg != "":
+                arg = self.resolve_value_or_variable(arg)
+                fixed_args.append(arg)
         if call in MODULE_FUNCTIONS.keys():
             func = MODULE_FUNCTIONS[call]
             try:
                 if not self.compile_only:
-                    self.last_value = func(arg)
+                    self.last_value = func(*fixed_args)

```

Comment:
Since now, both libraries and extensions can define **unlimited** arguments for functions.